ribs/archives/_cvt_archive.py

```diff
@@ -259,13 +264,36 @@ def index_of(self, measures_batch):
         if self._use_kd_tree:
             _, indices = self._centroid_kd_tree.query(measures_batch)
             return indices.astype(np.int32)
-
-        # Brute force distance calculation -- start by taking the difference
-        # between each measure i and all the centroids.
-        distances = np.expand_dims(measures_batch, axis=1) - self.centroids
-
-        # Compute the total squared distance -- no need to compute actual
-        # distance with a sqrt.
-        distances = np.sum(np.square(distances), axis=2)
-
-        return np.argmin(distances, axis=1).astype(np.int32)
+        else:
+            expanded_measures = np.expand_dims(measures_batch, axis=1)
+            # Compute indices chunks at a time
+            if (self._chunk_size is not None and
+                    self._chunk_size < measures_batch.shape[0]):
+                distances = np.sum(
```

Comment:
I just realized this is redundant with the for loop

Reply:
^ I think just having for-loop start at 1 will work.

Reply:
I also think the for-loop should start at 0 and instead of `i-1` just do `i`.

Reply:
See below comment on array_split